File: src/breslin/clientside/dispatch/dispatch.cpp
#include "dispatch.h"

#include "../tdreamsock/dreamSockLog.h"

// Common headers
#include <stdio.h>
#include <string.h>

Dispatch::Dispatch()
{
	mMaxSize = 1400;
	mCharArray = new char[mMaxSize];
	mSize		= 0;
	mReadCount	= 0;
}

Dispatch::~Dispatch()
{
}

/**********************************************
				ADMIN
************************************************/
void Dispatch::BeginReading()
{
	mReadCount = 0;
}

void Dispatch::Clear()
{
	mSize		= 0;
	mReadCount	= 0;
}

char *Dispatch::GetNewPoint(int length)
{
	char *tempData;

	// Check for overflow
	if(mSize + length > mMaxSize)
	{
		Clear(); 
	}
	tempData = mCharArray + mSize;
	mSize += length;

	return tempData;
}


/**********************************************
				WRITE
************************************************/
void Dispatch::WriteByte(char c)
{
	char *buf;
	buf = GetNewPoint(1);
	memcpy(buf, &c, 1);
}

void Dispatch::WriteShort(short c)
{
	char *buf;
	buf = GetNewPoint(2);
	memcpy(buf, &c, 2);
}

void Dispatch::WriteFloat(float c)
{
	char *buf;
	buf = GetNewPoint(4);
	memcpy(buf, &c, 4);
}

/**********************************************
				READ
************************************************/

char Dispatch::ReadByte()
{
	char c;

	if(mReadCount+1 > mSize)
		c = -1;
	else
		memcpy(&c, &mCharArray[mReadCount], 1);

	mReadCount++;

	return c;
}

short Dispatch::ReadShort()
{
	short c;

	if(mReadCount+2 > mSize)
		c = -1;
	else		
		memcpy(&c, &mCharArray[mReadCount], 2);

	mReadCount += 2;

	return c;
}

float Dispatch::ReadFloat()
{
	float c;

	if(mReadCount+4 > mSize)
		c = -1;
	else
		memcpy(&c, &mCharArray[mReadCount], 4);

	mReadCount += 4;

	return c;
}

File: src/breslin/clientside/dispatch/Dispatch.java

package breslin.clientside.dispatch;

/******************************************************
*				CLASS
********************************************************/
public class Dispatch
{


public Dispatch()
{
	mMaxSize = 1400;
	mCharArray = new byte[mMaxSize];
	mSize		= 0;
	mReadCount	= 0;
}
/**************************************************
*			VARIABLES
**************************************************/

	int				mMaxSize;
	int				mSize;
	int				mReadCount;
	byte[]			mCharArray;

/**************************************************
*			METHODS
**************************************************/
	//admin

int             getReadCount()

{
	return mReadCount;
}

void			BeginReading()
{

}

//char			*GetNewPoint(int length);

void			Clear()
{

}

int				GetSize()
{
	return mSize;
}

void			SetSize(int s)
{
	mSize = s;
}

//write
void			WriteByte  (byte  b)
{

}

void			WriteShort (short s)
{

}

void			WriteFloat (float f)
{

}

//read
byte			ReadByte   ()
{
	return 0;

}

short			ReadShort  ()
{
	return 0;
}
float			ReadFloat  ()
{
	return 0;
}




}